l10n_ch_payment_slip/payment_slip.py

```diff
@@ -480,10 +481,30 @@ def _draw_address(self, canvas, print_settings, initial_position, font,
         text.setFont(font.name, font.size)
         text.textOut(com_partner.name)
         text.moveCursor(0.0, font.size)
-        for line in com_partner.contact_address.split("\n"):
-            if not line:
-                continue
-            text.textLine(line)
+        pt_cpny_add = print_settings.bvr_print_cpny_address
+        
+        if pt_cpny_add == 'full':
+            for line in com_partner.contact_address.split("\n"):
+                if not line:
+                    continue
+                
+                text.textLine(line)
+        else:
+            city_line = (com_partner.zip or '') + ' ' + (com_partner.city or '') + ' ' + (com_partner.state_id.code or '')
+            if pt_cpny_add == 'first_line_only':
+                if com_partner.street:
+                    line = com_partner.street
+                    text.textLine(line)
+                    
+                text.textLine(city_line)
+                
+            if pt_cpny_add == 'second_line_only':
+                if com_partner.street2:
+                    line = com_partner.street2
+                    text.textLine(line)
+         
+                text.textLine(city_line)
```

Comment:
So in your database all your addresses have CP in `street` and actual address in `street2` ?

This configuration is system wide whereas you could have 10k of customer address without CP and 7k of customer address with CP. How do you work with both?

As for you to work with this case you have to put the actual address in `street2` I guess that you identified addresses with CP. This makes `street` the "case postale" field. Wouldn't it be better to make a new type of address name `bvr` in which you don't put the CP? Otherwise make a boolean on the address to know if the first line must be skipped.
Yet I don't see the option to choose street or street2 at system wide a good generic option. Maybe you could do that change in a specific module for your client just redefining `_draw_address`